### PR TITLE
Split machine shop sim

### PR DESCRIPTION
**Pull request by CourtneyJC**
Preface: Our master branch is currently broken, because of an issue with commits. Half of our changes exist on master, and the other half in the merging branch. We apologize for the confusion.

MachineShopSimulator.java runs the actual simulator as well as doing most of the setup. It should be split into separate classes. We have moved them into a new class called MachineShop, which handles the setup, leaving the simulation to MachineShopSimulator. We've also moved the changeState() method into Machine, as it makes more sense for a machine to handle its own states. This also makes it not just a data class.

### CODE REVIEW

File: src/applications/Machine.java [hunk 1]
```diff
@@ -104,4 +104,5 @@ public Job changeState() {// Task on theMachine has finished,
         return lastJob;
     }
 
+
```
Comment (bemica) [inline src/applications/Machine.java:107]: This new line makes the code MUCH more readable

**Review by bemica — APPROVED** [review]

File: src/applications/MachineShop.java [hunk 2]
```diff
@@ -39,15 +39,15 @@ static void startShop(SimulationSpecification specification) {
...
 
         for (int p = 1; p <= numMachines; p++)
-            changeState(p);
+            machineArray[p].changeState();
     }
 
     private static void createEventAndMachineQueues(SimulationSpecification specification) {
         // create event and machine queues
-        eList = new EventList(specification.getNumMachines(), MachineShopSimulator.largeTime);
+        eList = new EventList(specification.getNumMachines(), largeTime);
```
Comment (holma198) [inline src/applications/MachineShop.java:47]: I think that this is overall a better place to have time related variables, but the name 'largeTime' could probably have been named something a little better with this change.

**Comment by holma198** [comment]
### Code Review - RJ Holman

 ##### Documentation
Pull Request
The pull request seems lacking in detail as to why these changes were made.  Also there is no reference to which issues were closed either.  There are however 13 issues that exist as of reviewing.

 ##### Commits
Commits could occur more frequently.  This commit [63b4bcdbfc8ecc6414d4dd433cf96070c1a74de6], for instance has numerous changes in it that should be split up into smaller commits.  At least one for each file changed.  It is also not obvious as to why this was changed in the first place, be sure to make and include an issue in your commits!

 ##### refactoring quality
There were two main changes made to this repository.  Those changes included moving some `MachineShopSimulator` functionality into `MachineShop` a new class.  And there was also the change of moving some of the bookkeeping from `MachineShopSimulator` for machines into `Machine`.  The latter change is more straight forward.  The changes have enhanced the readability of the code and it seems logical to have the machine functionality in machine, which also addresses the issue of data classes.  Looking at the `MachineShopSimulator` changes, these I approve of as well.  Looking at `runSimulation` shows most of what was changed.  Overall both changes have enhanced readability of the code in a logical way.
 #### Bottom Line
 - Commit more frequently and add more specifics to commits
 - be sure to utilize the issues that you create in your commits
 - do NOT, commit changes to master in future commits.
 - Overall I would say to merge these changes into `master`, if for no other reason than to make it run again, however, focus on addressing these concerns in future commits and pull requests.
